react/PlaceOrder.tsx

```diff
@@ -1,14 +1,138 @@
+    rootPath = '',
+  } = useRuntime()
+
+  const [placingOrder, setPlacingOrder] = useState(false)
+  const [cardFormIframe] = useState(
+    () => document.getElementById('chk-card-form')! as HTMLIFrameElement
+  )
+
+  const handlePlaceOrder = async () => {
+    const transactionData = {
+      referenceId: orderForm.id,
+      value,
+      referenceValue,
+      interestValue,
+      savePersonalData: true,
+      optinNewsLetter: false,
+    }
+
+    setPlacingOrder(true)
+
+    const startTransactionResponse = await fetch(
+      `${rootPath}/api/checkout/pub/orderForm/${orderForm.id}/transaction`,
+      {
+        method: 'post',
+        body: JSON.stringify(transactionData),
+        headers: {
+          'content-type': 'application/json',
+        },
+      }
+    )
+
+    if (!startTransactionResponse.ok) {
+      setPlacingOrder(false)
+      return
+    }
+
+    const transaction = (await startTransactionResponse.json()) as TransactionResponse
+
+    const {
+      orderGroup: orderGroupId,
+      id: transactionId,
+      receiverUri,
+      merchantTransactions,
+      paymentData: { payments: transactionPayments },
+      gatewayCallbackTemplatePath,
+    } = transaction
+
+    if (merchantTransactions?.length > 0) {
+      const allPayments = transactionPayments.reduce<GatewayPayment[]>(
+        (payments, transactionPayment) => {
+          const merchantPayments = transactionPayment.merchantSellerPayments
+            .map(merchantPayment => {
+              const merchantTransaction = merchantTransactions.find(
+                merchant => merchant.id === merchantPayment.id
+              )
+
+              if (!merchantTransaction) {
+                return null
+              }
+
+              return {
+                ...transactionPayment,
+                ...merchantPayment,
+                currencyCode: currency as string,
+                transaction: {
+                  id: merchantTransaction.transactionId,
+                  merchantName: merchantTransaction.merchantName,
+                },
+              } as GatewayPayment
+            })
+            .filter(
+              (merchantPayment): merchantPayment is GatewayPayment =>
+                merchantPayment != null
+            )
+
+          return payments.concat(merchantPayments)
+        },
+        []
+      )
+
+      const { data: redirectUrl } = await postRobot.send(
+        cardFormIframe.contentWindow,
+        'sendPayments',
+        {
+          payments: allPayments,
+          receiverUri,
+          orderId: orderGroupId,
+          gatewayCallbackTemplatePath,
+          transactionId,
+        }
+      )
+
+      const callbackResponse = await fetch(
+        `${rootPath}/api/checkout/pub/gatewayCallback/${orderGroupId}`,
+        { method: 'POST' }
+      )
+
+      if (callbackResponse.ok) {
+        window.location.replace(redirectUrl)
+      }
+    } else if (!receiverUri) {
+      // go to first invalid step
```

Comment:
Will this condition receive code late on?

Reply:
yes! I just need to figure out what to do in this case 😂

the other sections in this if also needs to be updated, specially for handling error cases

Reply:
Ok. Do you have an issue/story documenting this?

Reply:
just created! [CHUI-49](https://vtex-dev.atlassian.net/browse/CHUI-49)